akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala

```diff
@@ -102,11 +105,26 @@ object ClusterShardingSettings {
       if (name == StateStoreModePersistence.name) StateStoreModePersistence
       else if (name == StateStoreModeDData.name) StateStoreModeDData
       else
-        throw new IllegalArgumentException("Not recognized StateStoreMode, only 'ddata' is supported.")
+        throw new IllegalArgumentException(
+          s"Not recognized StateStoreMode, only '${StateStoreModePersistence.name}' and '${StateStoreModeDData.name}' are supported.")
   }
   final case object StateStoreModePersistence extends StateStoreMode { override def name = "persistence" }
   final case object StateStoreModeDData extends StateStoreMode { override def name = "ddata" }
 
+  sealed trait RememberEntitiesStoreMode { def name: String }
+  object RememberEntitiesStoreMode {
+    def byName(name: String): RememberEntitiesStoreMode =
+      if (name == RememberEntitiesModeEventSourced.name) RememberEntitiesModeEventSourced
+      else if (name == RememberEntitiesStoreModeDData.name) RememberEntitiesStoreModeDData
+      else
+        throw new IllegalArgumentException(
+          s"Not recognized RememberEntitiesStore, only '${RememberEntitiesStoreModeDData.name}' and '${RememberEntitiesModeEventSourced.name}' are supported.")
+  }
+  final case object RememberEntitiesModeEventSourced extends RememberEntitiesStoreMode {
+    override def name = "eventsourced"
+  }
+  final case object RememberEntitiesStoreModeDData extends RememberEntitiesStoreMode { override def name = "ddata" }
```

Comment:
these objects are also not Java friendly

Reply:
They seem fine to use in type signatures from Java. Just can't create them.

I'll add factory methods